sympy/core/numbers.py

```diff
@@ -2422,6 +2422,69 @@ def __floordiv__(self, other):
     def __rfloordiv__(self, other):
         return Integer(Integer(other).p // self.p)
 
+    def __lshift__(self, other):
+        if isinstance(other, (int, Integer, numbers.Integral)):
+            return Integer(self.p << int(other))
+        else:
+            return NotImplemented
+
+    def __rlshift__(self, other):
+        if isinstance(other, (int, numbers.Integral)):
```

Comment:
Let me know if you think it's necessary/helpful to add `Integer` to this list. In the current implementation, I don't think it's necessary, and its omission signals the asymmetry of the `__r…__` methods. That being said, maybe there's some weird future where there's a `class Integer2(Integer): …` that overrides half of these methods such that these `__r…__` methods become "slow" again? Dunno.